.changeset/blue-pans-suffer.md

```diff
@@ -0,0 +1,6 @@
+---
+"@asyncapi/generator": minor
+---
+- Moved the logic of **conditionalFile generation** to a new **conditionalGeneration module**, which now supports both file and folder generation in a unified way.
```

Comment:
in below bullet points focus on end user that is reading release notes.

they are not interested in info that some logic/code was moved. The need to know what changed and if there is something to do, to react, they need clear instruction what to do:
- conditionalGeneration is added:
  - supports conditioning files and folders
  - supports conditioning basing on values from AsyncAPI document (`subject`) and parameters specified in a given generation template
- conditionalFile is deprecated and will be removed. To migrate, just rename to `conditionalGeneration`

of course above is quick example, needs better wording, use AI for that